omnibus/config/software/apt.rb

```diff
@@ -0,0 +1,96 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https:#www.datadoghq.com/).
+# Copyright 2016-present Datadog, Inc.
+
+require './lib/cmake.rb'
+
+name 'apt'
+default_version '2.5.5'
+
+dependency 'bzip2'
+dependency 'gnutls'
+dependency 'libdb'
+dependency 'libgcrypt'
+dependency 'libiconv'
+dependency 'liblz4'
+dependency 'liblzma'
+dependency 'libxxhash'
+dependency 'zstd'
+
+license 'GPLv2'
+license_file "COPYING"
+skip_transitive_dependency_licensing true
+
+version("2.5.5") { source sha256: "488d858485bd87369338cdbe3dcd74437379eebea9c9adf272df1e4a05714f3c" }
+
+source url: "https://github.com/Debian/apt/archive/refs/tags/#{version}.tar.gz"
+
+relative_path "apt-#{version}"
+
+build do
+  env = with_standard_compiler_flags(with_embedded_path)
+
+  env["PKG_CONFIG_PATH"] = "#{install_dir}/embedded/lib/pkgconfig"
+  env["CC"] = "/opt/gcc-#{ENV['GCC_VERSION']}/bin/gcc"
+  env["CXX"] = "/opt/gcc-#{ENV['GCC_VERSION']}/bin/g++"
+  env["CXXFLAGS"] += " -static-libstdc++ -std=c++11"
+  patch source: "no_doc.patch", env: env
+  patch source: "disable_arch_check.patch", env: env
```

Comment:
Not a blocker for now, but in the foreseeable future, could you please add comments to all the patches on all software definitions that explain why we have them? For example, it is not clear to me why we disable the arch check for APT.